src/Microsoft.DotNet.TestFramework/TestAssetsManager.cs

```diff
@@ -107,12 +107,48 @@ public TestDirectory CreateTestDirectory([CallerMemberName] string callingMethod
 
         private string GetTestDestinationDirectoryPath(string testProjectName, string callingMethod, string identifier)
         {
-#if NET451
+// HACK: Make the TestAssetsManager create shorter paths for test assets.
+// This is a workaround for https://github.com/dotnet/coreclr/issues/5046
+// Remove this hack once the above issue is fixed.
+/*#if NET451
             string baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
 #else
             string baseDirectory = AppContext.BaseDirectory;
+#endif*/
+
+            string path = Path.Combine(GetRepoRoot(), "bin", callingMethod + identifier, testProjectName);
+            Console.WriteLine($"[TestAssetsManager] The working directory for {testProjectName} is '{path}'");
+            return path;
+        }
+
+        private static string GetRepoRoot()
```

Comment:
@eerhardt This is going to be a temporary fix. This should go away once we get a coreclr build with the fix for https://github.com/dotnet/coreclr/issues/5046